q2cli/builtin/tools.py

```diff
@@ -290,7 +289,13 @@ def cast_metadata(paths, cast, output_file, ignore_extra,
     with tempfile.NamedTemporaryFile() as temp:
         metadata.save(temp.name)
         cast_md = qiime2.Metadata.load(temp.name, cast_dict)
+
+    if output_file:
         cast_md.save(output_file)
+    else:
```

Comment:
After doing some testing on the docs rendering, this actually won't work because it's not truly writing to stdout - so this is still a WIP! Working on this today, and will push up some changes after some more testing.